lib/fetch/util.js

```diff
@@ -332,8 +332,178 @@ function clonePolicyContainer () {
 
 // https://w3c.github.io/webappsec-referrer-policy/#determine-requests-referrer
 function determineRequestsReferrer (request) {
-  // TODO
-  return 'no-referrer'
+  // 1. Let policy be request's referrer policy.
+  const policy = request.referrerPolicy
+
+  // Return no-referrer when empty or policy says so
+  if (policy == null || policy === '' || policy === 'no-referrer') {
+    return 'no-referrer'
+  }
+
+  // 2. Let environment be the request client
+  const environment = request.client
+  let referrerSource = null
+
+  /**
+   * 3, Switch on request’s referrer:
+    "client"
+      If environment’s global object is a Window object, then
+        Let document be the associated Document of environment’s global object.
+        If document’s origin is an opaque origin, return no referrer.
+        While document is an iframe srcdoc document,
+        let document be document’s browsing context’s browsing context container’s node document.
+        Let referrerSource be document’s URL.
+
+      Otherwise, let referrerSource be environment’s creation URL.
+
+    a URL
+    Let referrerSource be request’s referrer.
+   */
+  if (request.referrer === 'client') {
+    // Not defined in Node but part of the spec
+    if (environment?.globalObject instanceof Window) { // eslint-disable-line
```

Comment:
This should be changed to something similar to https://github.com/nodejs/undici/blob/e5e5b97d83010cfb752eff8d9c8d488399cc3ab7/lib/fetch/index.js#L414

Reply:
Sure, applied in [6ed09c1](https://github.com/nodejs/undici/pull/1236/commits/6ed09c17be399e459f9174dbaf4fd30206bcb8c5)

Reply:
this doesn't seem to have been done

Reply:
True, sorry for that. Change was made but didn't stage it for the commit 😓 
Here it is the new one: 681a755